src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -155,21 +159,50 @@ public IStringLocalizer Create(string baseName, string location)
                 throw new ArgumentNullException(nameof(baseName));
             }
 
-            location = location ?? _hostingEnvironment.ApplicationName;
-
-            baseName = GetResourcePrefix(baseName, location);
+            location = location ?? _applicationName;
 
             return _localizerCache.GetOrAdd($"B={baseName},L={location}", _ =>
             {
-                var assembly = Assembly.Load(new AssemblyName(location));
-                return new ResourceManagerStringLocalizer(
-                    new ResourceManager(baseName, assembly),
-                    assembly,
-                    baseName,
-                    _resourceNamesCache);
+                var assemblyName = new AssemblyName(location);
+                var assembly = Assembly.Load(assemblyName);
+
+                baseName = GetResourcePrefix(baseName, location);
+                return CreateIStringLocalizer(assembly, baseName);
             });
         }
 
+        protected virtual ResourceManagerStringLocalizer CreateIStringLocalizer(Assembly assembly, string baseName)
+        {
+            return new ResourceManagerStringLocalizer(
+                new ResourceManager(baseName, assembly),
+                assembly,
+                baseName,
+                _resourceNamesCache);
+        }
+
+        protected virtual string GetResourcePrefix(string location, string baseName, string resourceLocation)
+        {
+            // Re-root the base name if a resources path is set
+            return location + "." + resourceLocation + TrimPrefix(baseName, location + ".");
+        }
+
+        protected virtual ResourceLocationAttribute GetResourceLocationAttribute(Assembly assembly)
```

Comment:
Not sure about the other `protected virtual` methods but please add a comment identifying this one as `virtual` for testing. Can the other two be `private` or (better) `private static`?